language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
```

Comment:
Similar here: this should never fail unless there's a bug in the analysis or `GlobalEnv`, so returning `RWOffset` makes sense.